internal/kafka/internal/services/quota/ams_quota_service_test.go

```diff
@@ -1000,6 +1001,195 @@ func Test_AMSCheckQuota(t *testing.T) {
+		Quota:                  config.NewKafkaQuotaConfig(),
+		SupportedInstanceTypes: test.NewAMSTestKafkaSupportedInstanceTypesConfig(),
+	}
+	type fields struct {
+		ocmClient   *ocm.ClientMock
+		kafkaConfig config.KafkaConfig
+	}
+	type args struct {
+		kafkaID                           string
+		kafkaInstanceType                 types.KafkaInstanceType
+		kafkaRequestMarketplace           string
+		kafkaRequestDesiredBillingModel   string
+		kafkaRequestBillingCloudAccountID string
+		kafkaRequestCloudProvider         string
+
+		wantsReserveCalled bool
+		wantsProductID     string
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		args   args
+	}{
+		{
+			name: "Reserve quota for new cluster",
+			fields: fields{
+				kafkaConfig: amsDefaultKafkaConf,
+				ocmClient: &ocm.ClientMock{
+					FindSubscriptionsFunc: func(query string) ([]*v1.Subscription, error) {
+						return ams.FindSubscriptions(query), nil
+					},
+					GetOrganisationIdFromExternalIdFunc: func(externalId string) (string, error) {
+						return fmt.Sprintf("fake-org-id-%s", externalId), nil
+					},
+					ClusterAuthorizationFunc: func(cb *v1.ClusterAuthorizationRequest) (*v1.ClusterAuthorizationResponse, error) {
+						sub := v1.SubscriptionBuilder{}
+						sub.ID(cb.ClusterID() + "subscription")
+						sub.Status("Active")
+						ca, _ := v1.NewClusterAuthorizationResponse().Allowed(true).Subscription(&sub).Build()
+						return ca, nil
+					},
+				},
+			},
+			args: args{
+				kafkaID:                         "1234",
+				kafkaInstanceType:               types.STANDARD,
+				kafkaRequestDesiredBillingModel: "EVAL",
+
+				wantsReserveCalled: true,
+				wantsProductID:     "RHOSAKEval",
+			},
+		},
+		{
+			name: "Reserve quota for existing cluster and existing subscription",
+			fields: fields{
+				kafkaConfig: amsDefaultKafkaConf,
+				ocmClient: &ocm.ClientMock{
+					FindSubscriptionsFunc: func(query string) ([]*v1.Subscription, error) {
+						return ams.FindSubscriptions(query), nil
+					},
+					GetReservedResourcesBySubscriptionIDFunc: func(subscriptionID string) ([]*v1.ReservedResource, error) {
+						return ams.GetReservedResourcesBySubscriptionID(subscriptionID), nil
+					},
+					GetOrganisationIdFromExternalIdFunc: func(externalId string) (string, error) {
+						return fmt.Sprintf("fake-org-id-%s", externalId), nil
+					},
+					ClusterAuthorizationFunc: func(cb *v1.ClusterAuthorizationRequest) (*v1.ClusterAuthorizationResponse, error) {
+						sub := v1.SubscriptionBuilder{}
+						sub.ID(cb.ClusterID() + "subscription")
+						sub.Status("Active")
+						ca, _ := v1.NewClusterAuthorizationResponse().Allowed(true).Subscription(&sub).Build()
+						return ca, nil
+					},
+				},
+			},
+			args: args{
+				kafkaID:                         "cluster-1",
+				kafkaInstanceType:               types.STANDARD,
+				kafkaRequestDesiredBillingModel: "STANDARD",
+
+				wantsReserveCalled: false,
+				wantsProductID:     "RHOSAK",
```

Comment:
The RHOSAKProduct constant available in the pkg/client/ocm package can be used instead